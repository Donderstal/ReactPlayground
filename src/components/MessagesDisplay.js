import * as React from 'react'
import {Platform, StyleSheet, Text, View, FlatList} from 'react-native';

const MessagesDisplay = (props) => {
    if (props.messages.length > 0) {
        return (
            <View style={{backgroundColor: 'lightblue', borderRightColor: '#f4f984', flex: .25, margin: '5%', padding: '5%',
            borderRightWidth: StyleSheet.hairlineWidth}}>
                 <FlatList
                    data={props.messages}
                    renderItem={({item}) => <Text>{item.text}</Text>}
                /> 
            </View>
        )        
    }
    else {
        return (
            <View style={{backgroundColor: 'lightblue', borderRightColor: '#f4f984', flex: .25, margin: '5%', padding: '5%',
            borderRightWidth: StyleSheet.hairlineWidth}}> 
                <Text>No data received yet...</Text>
            </View>
        )        
    }

}

export default MessagesDisplay