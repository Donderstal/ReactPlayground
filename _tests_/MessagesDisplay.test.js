import React from 'react';
import renderer from 'react-test-renderer';


import MessagesDisplay from '../src/components/MessagesDisplay';


test('test if mDisplay renders', () => {
     messages = ['Hey', 'this', 'is', 'a', 'test']
     const tree = renderer.create(<MessagesDisplay messages={messages}/>).toJSON();
      expect(tree).toMatchSnapshot();
});