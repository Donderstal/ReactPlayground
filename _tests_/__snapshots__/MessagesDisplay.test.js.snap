// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test if mDisplay renders 1`] = `
<View
  style={
    Object {
      "backgroundColor": "lightblue",
      "borderRightColor": "#f4f984",
      "borderRightWidth": 0.5,
      "flex": 0.25,
      "margin": "5%",
      "padding": "5%",
    }
  }
>
  <RCTScrollView
    data={
      Array [
        "Hey",
        "this",
        "is",
        "a",
        "test",
      ]
    }
    disableVirtualization={false}
    getItem={[Function]}
    getItemCount={[Function]}
    horizontal={false}
    initialNumToRender={10}
    keyExtractor={[Function]}
    maxToRenderPerBatch={10}
    numColumns={1}
    onContentSizeChange={[Function]}
    onEndReachedThreshold={2}
    onLayout={[Function]}
    onMomentumScrollEnd={[Function]}
    onScroll={[Function]}
    onScrollBeginDrag={[Function]}
    onScrollEndDrag={[Function]}
    renderItem={[Function]}
    scrollEventThrottle={50}
    stickyHeaderIndices={Array []}
    updateCellsBatchingPeriod={50}
    viewabilityConfigCallbackPairs={Array []}
    windowSize={21}
  >
    <View>
      <View
        onLayout={[Function]}
        style={null}
      >
        <Text />
      </View>
      <View
        onLayout={[Function]}
        style={null}
      >
        <Text />
      </View>
      <View
        onLayout={[Function]}
        style={null}
      >
        <Text />
      </View>
      <View
        onLayout={[Function]}
        style={null}
      >
        <Text />
      </View>
      <View
        onLayout={[Function]}
        style={null}
      >
        <Text />
      </View>
    </View>
  </RCTScrollView>
</View>
`;
